pkg/traceability/sampling/globalsampling.go

```diff
@@ -18,28 +25,36 @@ import (
 var agentSamples *sample
 
 const (
-	qaSamplingPercentageEnvVar = "QA_TRACEABILITY_SAMPLING_PERCENTAGE"
+	qaSamplingPercentageEnvVar   = "QA_TRACEABILITY_SAMPLING_PERCENTAGE"
+	qaErrorSamplingResetInterval = "QA_TRACEABILITY_ERROR_SAMPLING_RESET_INTERVAL"
 )
 
 // Sampling - configures the sampling of events the agent sends to Amplify
 type Sampling struct {
-	Percentage      float64 `config:"percentage"`
-	PerAPI          bool    `config:"per_api"`
-	PerSub          bool    `config:"per_subscription"`
-	OnlyErrors      bool    `config:"onlyErrors" yaml:"onlyErrors"`
-	countMax        int
-	shouldSampleMax int
+	Percentage                 float64 `config:"percentage"`
+	PerAPI                     bool    `config:"per_api"`
+	PerSub                     bool    `config:"per_subscription"`
+	OnlyErrors                 bool    `config:"onlyErrors" yaml:"onlyErrors"`
+	ErrorSamplingEnabled       bool    `config:"errorSamplingEnabled"`
+	ErrorSamplingResetInterval time.Duration
+	countMax                   int
+	shouldSampleMax            int
+	cacheManager               cache.Manager
+	externalDataLookUp         map[string]string
 }
 
 // DefaultConfig - returns a default sampling config where all transactions are sent
 func DefaultConfig() Sampling {
 	return Sampling{
-		Percentage:      defaultSamplingRate,
-		PerAPI:          true,
-		PerSub:          true,
-		OnlyErrors:      false,
-		countMax:        countMax,
-		shouldSampleMax: defaultSamplingRate,
+		Percentage:                 defaultSamplingRate,
+		PerAPI:                     true,
+		PerSub:                     true,
+		OnlyErrors:                 false,
+		ErrorSamplingEnabled:       false,
+		ErrorSamplingResetInterval: 1 * time.Hour,
```

Comment:
can we read the qa variable here rather than getErrorSamplingResetIntervalConfig is